tests/core/full_node/full_sync/test_full_sync.py

```diff
@@ -412,3 +415,59 @@ async def test_sync_none_wp_response_backward_comp(self, three_nodes, default_10
         assert res is None
         duration = time.time() - start
         assert duration > 5
+
+    @pytest.mark.asyncio
+    async def test_bad_peak_cache(
+        self, three_nodes, default_400_blocks, blockchain_constants, self_hostname, consensus_mode, bt
+    ):
+        if consensus_mode != Mode.PLAIN:
+            pytest.skip("Skipped test")
+        full_node_1, full_node_2, full_node_3 = three_nodes
+        bt.constants = blockchain_constants.replace(SOFT_FORK4_HEIGHT=1000000)
+        blocks = bt.get_consecutive_blocks(700, default_400_blocks)
+        full_node_2.full_node.blockchain.constants = blockchain_constants.replace(SOFT_FORK4_HEIGHT=1000000)
+        full_node_1.full_node.blockchain.constants = blockchain_constants.replace(SOFT_FORK4_HEIGHT=400)
+        for block in blocks:
+            await full_node_2.full_node.add_block(block)
+        server_1 = full_node_1.full_node.server
+        server_2 = full_node_2.full_node.server
+        await server_2.start_client(
+            PeerInfo(self_hostname, uint16(server_1._port)), on_connect=full_node_2.full_node.on_connect
+        )
+        await time_out_assert(60, node_height_between, True, full_node_1, 380, 450)
+        await time_out_assert(60, full_node_1.full_node.sync_store.get_long_sync, False)
+        # call peer has block to populate peer_to_peak
+        peak = full_node_2.full_node.blockchain.get_peak()
+        wp = await full_node_2.full_node.weight_proof_handler.get_proof_of_weight(peak.header_hash)
+        assert full_node_1.full_node.check_bad_peak_cache(wp) is False
+
+    @pytest.mark.asyncio
+    async def test_skip_bad_peak_validation(
+        self, three_nodes, default_400_blocks, blockchain_constants, self_hostname, consensus_mode, bt
```

Comment:
hints

Reply:
really ? this whole file is unhinted and is not checked by mypy...

Reply:
this whole file is unhinted and is not checked by mypy...

i dont think there is any point in doing it if not doing the whole file and removing from exemptions, and im not going to do that as part of this pr

Reply:
The point is that presently the test isn't type checked at all since the function parameters and return are not hinted.  If they were hinted, the contents would get checked, albeit with the looser mypy constraints as you note.

But, I'll walk away from this.  You can close this conversation after reading the note.